component/src/main/java/org/wso2/extension/siddhi/execution/env/ResourceBatchWindowProcessor.java

```diff
@@ -0,0 +1,383 @@
+                            nextEmitTime = newNextEmitTime;
+                            if (scheduler != null) {
+                                scheduler.notifyAt(nextEmitTime);
+                            }
+                        }
+                    }
+                }
+                if (outputExpectsExpiredEvents) {
+                    for (Map.Entry<Object, ResourceStreamEventList> entry : groupEventMap.entrySet()) {
+                        windowLength = ResourceIdentifierStreamProcessor.getResourceCount(resourceName);
+                        if (entry.getValue().streamEventList.size() >= windowLength ||
+                                entry.getValue().expiryTimestamp < currentTime) {
+                            //update current event chunk with event batch
+                            for (StreamEvent event : entry.getValue().streamEventList) {
+                                currentEventChunk.add(event);
+                            }
+                            groupEventMap.remove(entry.getKey());
+                            //update outputStreamEventChunk with expired events chunk
+                            if (eventsToBeExpired.getFirst() != null) {
+                                while (eventsToBeExpired.hasNext()) {
+                                    StreamEvent expiredEvent = eventsToBeExpired.next();
+                                    expiredEvent.setTimestamp(currentTime);
+                                }
+                                outputStreamEventChunk.add(eventsToBeExpired.getFirst());
+                            }
+                            eventsToBeExpired.clear();
+                            //update outputStreamEventChunk and eventsToBeExpired with current event chunk
+                            if (currentEventChunk.getFirst() != null) {
+                                // add reset event in front of current events
+                                outputStreamEventChunk.add(resetEvent);
+                                resetEvent = null;
+                                currentEventChunk.reset();
+                                while (currentEventChunk.hasNext()) {
+                                    StreamEvent currentEvent = currentEventChunk.next();
+                                    StreamEvent toExpireEvent = streamEventCloner.copyStreamEvent(currentEvent);
+                                    toExpireEvent.setType(StreamEvent.Type.EXPIRED);
+                                    eventsToBeExpired.add(toExpireEvent);
+                                }
+                                resetEvent = streamEventCloner.copyStreamEvent(eventsToBeExpired.getFirst());
+                                resetEvent.setType(ComplexEvent.Type.RESET);
+                                outputStreamEventChunk.add(currentEventChunk.getFirst());
+                            }
+                            currentEventChunk.clear();
+                        }
```

Comment:
Rewrite the about as 
```
if (streamEventList!=null) {
  if (outputExpectsExpiredEvents) {
      //for each event in streamEventList
      //    clone that and make it expired and add that to eventsToBeExpired.
      //end for loop 
  }
// get the first event of streamEventList clone that, make it as reset event, 
// and that and add that to eventsToBeExpired.
}
```

Reply:
Fixes added: d6242876caf0f65c7fd023c25c7c5e50e9fd7b3c